packages/sdk-v2/src/sdk/savings/react.ts

```diff
@@ -0,0 +1,174 @@
+}
+
+export const useGlobalStats = (refresh: QueryParams["refresh"] = "never", env?: EnvKey) => {
+const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
+
+const results = useCalls(
+  [
+    {
+      contract: gdStaking,
+      method: 'stats',
+      args: [],
+    },
+    {
+      contract: gdStaking,
+      method: 'getRewardsPerBlock',
+      args: []
+    },
+    {
+      contract: gdStaking,
+      method: 'numberOfBlocksPerYear',
+      args: []
+    }
+  ],{ refresh });
+
+  let globalStats:GlobalStats = {
+    totalStaked: undefined,
+    totalRewardsPaid: undefined,
+    apy: undefined
+  }
+
+
+  if (results[0]?.error){ // one fails, all fails
+    let errMessages: Array<any> = [];
+    for (let i = 0; i < results.length; i++){
+      errMessages.push(results[i]?.error)
+    }
+
+    return {
+      stats: undefined,
+      error: errMessages
+    }
+  }
+
+  if (results[0]){
+    const [lastUpdateBlock, totalStaked, totalRewardsPaid, savings] = results[0]?.value
+    const staked = CurrencyAmount.fromRawAmount(G$[122], totalStaked)
+    const rewardsPaid = CurrencyAmount.fromRawAmount(G$[122], totalRewardsPaid)
+    globalStats.totalStaked = staked
+    globalStats.totalRewardsPaid = rewardsPaid
+  }
+
+  if (results[1] && results[2]){
+    const {_goodRewardPerBlock: grpb, _gdInterestRatePerBlock: gdIrpb} = results[1]?.value
+    const numberOfBlocksPerYear = results[2]?.value
+    const apy = (Math.pow((gdIrpb/1e18), numberOfBlocksPerYear) - 1) * 100
+
+    globalStats.apy = apy
+  }
+
+  return {
+    stats: globalStats,
+    error: undefined
+  }
+}
+
+export const useStakerInfo = (refresh: QueryParams["refresh"] = "never", account:string, env?: EnvKey) => {
+  const contract = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking
+  const results = useCalls(
+    [
+      {
+        contract: contract,
+        method: 'getUserPendingReward(address)',
+        args: [account]
+      },
+      {
+        contract: contract,
+        method: 'principle',
+        args: [account]
+      },
+    ], { refresh }
+  );
+
+  let stakerInfo:StakerInfo = {
+    claimable: undefined,
+    principle: undefined
+  }
+
+  if (results[0]?.error){
+    let errMessages: Array<any> = [];
+    for (let i = 0; i < results.length; i++){
+      errMessages.push(results[i]?.error)
+    }
+
+    return {
+      stats: undefined,
+      error: errMessages
+    }
+  }
+  
+  if (results[0]){
+    const [goodReward, g$Reward ] = results[0]?.value
+    const claimableRewards = {
+      g$Reward: CurrencyAmount.fromRawAmount(G$[122], g$Reward),
+      goodReward: CurrencyAmount.fromRawAmount(G$[122], goodReward).divide(1e16)
+    }
+    stakerInfo.claimable = claimableRewards
+  }
+
+  if (results[1]){
+    const [principle] = results[1]?.value; //note: original deposit
+    const deposit = CurrencyAmount.fromRawAmount(G$[122], principle)
```

Comment:
you have 122 hard coded, what if we use savings on Celo?

Reply:
Updated